src/Microsoft.ML.Data/Scorers/PredictionTransformer.cs

```diff
@@ -74,16 +69,72 @@ internal PredictionTransformerBase(IHost host, ModelLoadContext ctx)
             ms.Position = 0;
             var loader = new BinaryLoader(host, new BinaryLoader.Arguments(), ms);
             TrainSchema = loader.Schema;
+        }
+
+        public abstract ISchema GetOutputSchema(ISchema inputSchema);
+
+        /// <summary>
+        /// Transforms the input data.
+        /// </summary>
+        /// <param name="input"></param>
+        /// <returns></returns>
+        public abstract IDataView Transform(IDataView input);
 
+        protected void SaveModel(ModelSaveContext ctx)
+        {
+            // *** Binary format ***
+            // model: prediction model.
+            // stream: empty data view that contains train schema.
+            // id of string: feature column.
+
+            ctx.SaveModel(Model, DirModel);
+            ctx.SaveBinaryStream(DirTransSchema, writer =>
+            {
+                using (var ch = Host.Start("Saving train schema"))
+                {
+                    var saver = new BinarySaver(Host, new BinarySaver.Arguments { Silent = true });
+                    DataSaverUtils.SaveDataView(ch, saver, new EmptyDataView(Host, TrainSchema), writer.BaseStream);
+                }
+            });
+        }
+    }
+
+        public abstract class ClassicPredictionTransformerBase<TModel> : PredictionTransformerBase<TModel>, IClassicPredictionTransformer<TModel>, ICanSaveModel
```

Comment:
I guess by "classic" this just means a prediction transformer base that takes a single features column as its input. Classic is a bit of a funny word, but then again `SinlgeFeaturesPredictionTransformerBase` might be a bit of a mouthful and itself potentially confusing.